### PR TITLE
Implimenting hooks validation

### PR DESCRIPTION
**Pull request by gangotriRails**
- installing react-hooks-forms in client side
- using hooks validation for address form

### CODE REVIEW

File: client/src/components/Address/AddressForm.js [hunk 1]
```diff
@@ -1,132 +1,229 @@
...
+          id: id,
+          name: data.name,
+          email: data.email,
+          phone: data.phone,
+          street: data.street,
+          city: data.city,
+          pin_code: data.pin_code,
+          state: data.state,
+          country: country.label,
+          user_id: user_id,
```
Comment (iMagesh) [inline client/src/components/Address/AddressForm.js:94]: simplify these as well?

File: client/src/components/Address/AddressForm.js [hunk 2]
```diff
@@ -1,132 +1,229 @@
...
+    //       setPhone(response.data.phone);
+    //       setName(response.data.name);
+    //       setEmail(response.data.email);
+    //       setTitle("Address Edit:");
+    //       SetIsCountry(true);
+    //     })
+    //     .catch((err) => {
+    //       setMessage("Sorry we couldnot get address with error " + err);
+    //     });
+    // }
```
[the remark this answers is not in the document]
Comment (Gangotri-sg3) [reply client/src/components/Address/AddressForm.js]: removed

File: client/src/components/Address/AddressForm.js [hunk 3]
```diff
@@ -1,132 +1,229 @@
...
+      .post("/user/new/address/", {
+        name: data.name,
+        email: data.email,
+        phone: data.phone,
+        street: data.street,
+        city: data.city,
+        pin_code: data.pin_code,
+        state: data.state,
+        country: country.label,
+        user_id: user_id,
```
[the remark this answers is not in the document]
Comment (Gangotri-sg3) [reply client/src/components/Address/AddressForm.js]: resolved

File: client/src/components/Address/AddressForm.js (hunk 1 above)
Comment (Gangotri-sg3) [reply client/src/components/Address/AddressForm.js]: resolved

**Comment by iMagesh** [comment]
@Gangotri-sg3 resolve the conficts?

**Comment by Gangotri-sg3** [comment]
@iMagesh resolved conflicts

File: server/routes/order/address.js [hunk 4]
```diff
@@ -31,13 +31,13 @@ router.get("/:id", async (req, res, next) => {
...
-      state: req.body.state,
+      name: req.body.data.name,
+      email: req.body.data.email,
+      phone: req.body.data.phone,
+      street: req.body.data.street,
+      city: req.body.data.city,
+      pin_code: req.body.data.pin_code,
+      state: req.body.data.state,
       country: req.body.country,
       user_id: req.body.user_id,
```
Comment (iMagesh) [inline server/routes/order/address.js:42]: can you simplify this?

File: server/routes/order/address.js [hunk 5]
```diff
@@ -99,13 +99,13 @@ router.put("/:id/edit", (req, res, next) => {
...
-            pin_code: req.body.pin_code,
-            state: req.body.state,
+            name: req.body.data.name,
+            email: req.body.data.email,
+            phone: req.body.data.phone,
+            street: req.body.data.street,
+            city: req.body.data.city,
+            pin_code: req.body.data.pin_code,
+            state: req.body.data.state,
             country: req.body.country,
```
Comment (iMagesh) [inline server/routes/order/address.js:109]: simplify this too?

**Comment by iMagesh** [comment]
@Gangotri-sg3 did you resolve the comments ?

File: server/routes/order/address.js (hunk 4 above)
Comment (Gangotri-sg3) [reply server/routes/order/address.js:42]: This is needed ,Don't need to simplify I guess @iMagesh Please check

**Comment by iMagesh** [comment]
I'm not merging this code because now we are using Formik instead. Please create another branch and use Formik + Yup for the validation. Check all the forms and see if it has been integrated, if not, you can do it and send a PR.

**Pull request closed without merging**